File: Src/Icm.Core/Math/STAT.vb
Namespace Icm.MathTools
    ''' <summary>
    '''  Module for easy statistical data generation. Not safe for multithreading,
    ''' though, you should use one StatisticsGenerator for each thread in order to
    ''' avoid problems.
    ''' </summary>
    ''' <remarks></remarks>
    Public Module STAT
        Private shrng As New Random

        Sub ChangeSeed(ByVal newseed As Integer)
            shrng = New Random(newseed)
        End Sub

        Sub New()
            shrng = New Random
        End Sub

        Function Uniform01() As Double

            Dim result As Double = shrng.NextDouble()

            Return result
        End Function

        Function Uniform(ByVal min As Double, ByVal max As Double) As Double
            Return Uniform01() * (max - min) + min
        End Function

    End Module
End Namespace

File: Src/Icm.Core/Math/STAT.cs
using System;

namespace Icm.MathTools
{
	/// <summary>
	///  Module for easy statistical data generation. Not safe for multithreading,
	/// though, you should use one <see cref="StatisticsGenerator"></see> for each thread in order to
	/// avoid problems.
	/// </summary>
	/// <remarks></remarks>
	public static class STAT
	{

		private static Random shrng = new Random();
		public static void ChangeSeed(int newseed)
		{
			shrng = new Random(newseed);
		}

		public STAT()
		{
			shrng = new Random();
		}

		public static double Uniform01()
		{

			double result = shrng.NextDouble();

			return result;
		}

		public static double Uniform(double min, double max)
		{
			return Uniform01() * (max - min) + min;
		}

	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
